eng/scripts/Cadl-Project-Generate.ps1

```diff
@@ -0,0 +1,58 @@
+[CmdletBinding()]
+param (
+    [Parameter(Position=0)]
+    [ValidateNotNullOrEmpty()]
+    [string] $ProjectDirectory
+)
+
+$ErrorActionPreference = "Stop"
+. $PSScriptRoot/../common/scripts/Helpers/PSModule-Helpers.ps1
+Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
+
+function NpmInstallForProject([string]$workingDirectory) {
+    Push-Location $workingDirectory
+    try {
+        $currentDur = Resolve-Path "."
+        Write-Host "Generating from $currentDur"
+        if (Test-Path "package.json") {
+            Remove-Item -Path "package.json" -Force
+        }
+        if (Test-Path ".npmrc") {
+            Remove-Item -Path ".npmrc" -Force
+        }
+        Copy-Item -Path "$PSScriptRoot/../csharp-emitter-package.json" -Destination "package.json" -Force
+        npm install
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+$cadlConfigurationFile = Resolve-Path "$ProjectDirectory/src/cadl-location.yaml"
+
+Write-Host "Reading configuration from $cadlConfigurationFile"
+$configuration = Get-Content -Path $cadlConfigurationFile -Raw | ConvertFrom-Yaml
+
+$specSubDirectory = $configuration["directory"]
+$innerFolder = Split-Path $specSubDirectory -Leaf
+
+$tempFolder = "$ProjectDirectory/TempCadlFiles"
+$npmWorkingDir = Resolve-Path $tempFolder/$innerFolder
+$mainCadlFile = Resolve-Path "$npmWorkingDir/main.cadl"
+$resolvedProjectDirectory = Resolve-Path "$ProjectDirectory/src"
+
+try {
+    Push-Location $npmWorkingDir
+    NpmInstallForProject $npmWorkingDir
+    Write-Host("npx cadl compile $mainCadlFile --emit `"`@azure-tools/cadl-csharp`" --output-path `"$resolvedProjectDirectory`" --option @azure-tools/cadl-csharp.clear-output-folder=true")
+    npx cadl compile $mainCadlFile --emit "@azure-tools/cadl-csharp" --output-path "$resolvedProjectDirectory" --option @azure-tools/cadl-csharp.clear-output-folder=true
```

Comment:
Why the options are hard-coded here? Shouldn't it be in the cadl-project.yaml?

Reply:
And my understanding is there should be logic some where to move all the parts only leaving "@azure-tools/cadl-csharp" part. Is this feature included?